### ISSUE TITLE
consolidate image names

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/docker-images#293]
As of #287 / #290 / #291, we'll have the following images

```
- quay.io/condaforge/linux-anvil-cos7-x86_64        # cos7
- quay.io/condaforge/linux-anvil-aarch64            # cos7
- quay.io/condaforge/linux-anvil-ppc64le            # cos7
- quay.io/condaforge/linux-anvil-alma-x86_64:8      # alma8
- quay.io/condaforge/linux-anvil-alma-aarch64:8     # alma8
- quay.io/condaforge/linux-anvil-alma-ppc64le:8     # alma8
- quay.io/condaforge/linux-anvil-alma-x86_64:9      # alma9
- quay.io/condaforge/linux-anvil-alma-aarch64:9     # alma9
- quay.io/condaforge/linux-anvil-alma-ppc64le:9     # alma9

- quay.io/condaforge/linux-anvil-cuda:11.8          # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-aarch64-cuda:11.8  # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-ppc64le-cuda:11.8  # ubi8 + CUDA 11.8
```

I think it would be nice to could consolidate this to (updated!; v0 below)
```
- quay.io/condaforge/linux-anvil-x86_64:{cos7,alma8,alma9}
- quay.io/condaforge/linux-anvil-aarch64:{cos7,alma8,alma9}
- quay.io/condaforge/linux-anvil-ppc64le:{cos7,alma8,alma9}

- quay.io/condaforge/linux-anvil-x86_64-cuda:11.8   # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-aarch64-cuda:11.8  # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-ppc64le-cuda:11.8  # ubi8 + CUDA 11.8
```

This has the advantage that we don't have to duplicate the specification in the pinning across image versions, because we can just directly insert `DEFAULT_LINUX_VERSION` through jinja (see comment below).

<details>
<summary>Old proposal</summary>

```
- quay.io/condaforge/linux-anvil-x86_64:{7,8,9}     # cos7 / alma8 / alma9
- quay.io/condaforge/linux-anvil-aarch64:{7,8,9}    # cos7 / alma8 / alma9
- quay.io/condaforge/linux-anvil-ppc64le:{7,8,9}    # cos7 / alma8 / alma9

- quay.io/condaforge/linux-anvil-x86_64-cuda:11.8   # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-aarch64-cuda:11.8  # ubi8 + CUDA 11.8
- quay.io/condaforge/linux-anvil-ppc64le-cuda:11.8  # ubi8 + CUDA 11.8
```

This is because the introduction of another distro for the same RHEL generation (e.g. rocky8 next to alma8) is completely unrealistic at the moment, so there's no need to encode the distro in the image name. And if that time ever comes, we could still rename the images again.

---
</details>

If we want to encode the distro version also in the CUDA images, we could do
```
- quay.io/condaforge/linux-anvil-x86_64-cuda11.8:{cos7,ubi8}
- quay.io/condaforge/linux-anvil-aarch64-cuda11.8:ubi8
- quay.io/condaforge/linux-anvil-ppc64le-cuda11.8:ubi8
```
but I think this is not necessary because the CUDA 11.8 images are on the way out. It would only become relevant IMO if some dependency starts requiring `__glibc>=2.34` while we're still supporting 11.8 (though we could also just move the CUDA 11.8 images to ubi9 universally in that case, c.f. https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/6283). Note that there are no ubi9 [images](https://hub.docker.com/r/nvidia/cuda/tags?name=11.8.0) yet (and neither are there cos7 images for aarch/ppc).

**Comment by h-vetinari** [comment]
I wanted to see what this looked like in the pinning, so I opened a demo [PR](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6687). However, while putting that together, I realized that we can do even better, by using the same image tags as we use for our distro-naming, so we can just directly insert `DEFAULT_LINUX_VERSION`. This drastically shortens the specification in the

```yaml
docker_image:   # [os.environ.get("BUILD_PLATFORM", "").startswith("linux-")]
  # images for non-CUDA-enabled builds
  - quay.io/condaforge/linux-anvil-x86_64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}   # [os.environ.get("BUILD_PLATFORM") == "linux-64"]
  - quay.io/condaforge/linux-anvil-aarch64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}  # [os.environ.get("BUILD_PLATFORM") == "linux-aarch64"
  - quay.io/condaforge/linux-anvil-ppc64le:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}  # [os.environ.get("BUILD_PLATFORM") == "linux-ppc64le"]

  # images for CUDA 11.8 builds (no choice via DEFAULT_LINUX_VERSION available)
  - [omitted]

  # images for CUDA 12 builds
  - quay.io/condaforge/linux-anvil-x86_64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}   # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM") == "linux-64"]
  # case: native compilation (build == target)
  - quay.io/condaforge/linux-anvil-aarch64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}  # [aarch64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM") == "linux-aarch64"]
  - quay.io/condaforge/linux-anvil-ppc64le:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}  # [ppc64le and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM") == "linux-ppc64le"]
  # case: cross-compilation (build != target)
  - quay.io/condaforge/linux-anvil-x86_64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}   # [aarch64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM") == "linux-64"]
  - quay.io/condaforge/linux-anvil-x86_64:{{ environ.get("DEFAULT_LINUX_VERSION", "alma9") }}   # [ppc64le and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM") == "linux-64"]
```

I've updated the OP to reflect that proposal. This also has the advantage that if we ever do have different distros per generation, the image tag can absorb that distinction.

**Comment by h-vetinari** [comment]
Following the core call today, we ended up with the following set
```
- quay.io/condaforge/linux-anvil-x86_64:{cos7,alma8,alma9}
- quay.io/condaforge/linux-anvil-aarch64:{cos7,alma8,alma9}
- quay.io/condaforge/linux-anvil-ppc64le:{cos7,alma8,alma9}

- quay.io/condaforge/linux-anvil-x86_64-cuda11.8:{cos7,ubi8}
- quay.io/condaforge/linux-anvil-aarch64-cuda11.8:ubi8
- quay.io/condaforge/linux-anvil-ppc64le-cuda11.8:ubi8
```

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

This seems reasonable to me

Would it make sense to have a news entry with this change? Asking in case others are using our images and will need to make updates

**Comment by h-vetinari** [comment]
> Would it make sense to have a news entry with this change?

The old images will still be available, but not a problem to write a short update for this once #291 lands.

**Comment by h-vetinari** [comment]
> not a problem to write a short update for this once [#291](https://github.com/conda-forge/docker-images/pull/291) lands.

I think it'll be worthwhile anyway to write an update when the default image changes, which would also provide a canonical place where we can tell people to stay on the oldest image for feedstocks that are doing binary repackaging.

**Comment by jakirkham** [comment]
Certainly though sometimes folks don't notice and pick up the old image by accident ( for example: https://github.com/conda-forge/docker-images/issues/277#issuecomment-2246688114 )

Indeed. Think it makes sense to have one news entry for all of these changes. As noted the old images will be around so we can do this after making the changes if that is easier

**Comment by h-vetinari** [comment]
Closed by #291
